### PR TITLE
update storage service to also use s3 bucket

### PR DESCRIPTION
**Pull request by tonychang04**
1. Modified storage backend service, if we have s3bucket, we use multitennat structure with app key, else we use local storage
2. Installed aws sdk
3. modifed tests and added cloud tests.
4. rewrite tests/local/test-traditional-rest.sh as that is not writtedn correctly

Screenshot 2025-08-01 at 10 45 20

The video is too large such that i can't include it...

npm run test:e2e

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -69,1681 +70,3258 @@
         "node": ">=6.0.0"
       }
     },
-    "node_modules/@babel/helper-string-parser": {
```
Comment (tonychang04) [inline package-lock.json:72]: Package lock is required as we are using aws-sdk now

**Review by Fermionic-Lyu — APPROVED** [review]
Local test passed on my end. Let's merge and further implment & test s3 features

**Comment by tonychang04** [comment]
+1, we definitely will test again and again for our cloud solution for edge cases! Let me know if you want to sync up of cloud testing or if you are unsure about how to setup key locally. @Fermionic-Lyu .

Merging for now.
